### PR TITLE
Feature/fast import

### PR DESCRIPTION
**Pull request by qqmbr4k**

### CODE REVIEW

File: rest-server/core/src/main/scala/com/actionml/core/backup/FSMirror.scala [hunk 1]
```diff
@@ -129,4 +114,27 @@ class FSMirror(mirrorContainer: String, engineId: String)
     }
     Valid(jsonComment("Job created to import events in the background."))
   }
+
+  private def importFromFile(file: File, engine: Engine): Long = {
+    var eventsProcessed = 0
+    val src = Source.fromFile(file)
+    try {
+      src.getLines().sliding(512, 512).foreach { lines =>
```
Comment (andreyshikov) [inline rest-server/core/src/main/scala/com/actionml/core/backup/FSMirror.scala:122]: @pferrel do we need that setting?

Comment (pferrel) [reply rest-server/core/src/main/scala/com/actionml/core/backup/FSMirror.scala:122]: No idea. What is this setting based on? Does the `Source.fromFile` library not calculate a reasonable default? Did it speed up the import to set this `(512, 512)`???

@andreyshikov

File: rest-server/core/src/main/scala/com/actionml/core/engine/Engine.scala [hunk 2]
```diff
@@ -126,9 +126,11 @@ abstract class Engine extends LazyLogging with JsonSupport {
   def input(json: String): Validated[ValidateError, Response] = {
     // flatten the event into one string per line as per Spark json collection spec
     mirroring.mirrorEvent(json.replace("\n", " ") + "\n")
-    Valid(Comment("Input processed by base Engine"))
+      .andThen(_ => Valid(Comment("Input processed by base Engine")))
   }
 
+  def inputMany: Seq[String] => Unit = _.foreach(input)
```
Comment (pferrel) [inline rest-server/core/src/main/scala/com/actionml/core/engine/Engine.scala:132]: Is this API extension important? It seems to only avoid one input per call. The call overhead should not be important, is it?

@andreyshikov

File: rest-server/engines/src/main/scala/com/actionml/engines/ur/URDataset.scala [hunk 3]
```diff
@@ -163,6 +153,44 @@ class URDataset(engineId: String, val store: Store) extends Dataset[UREvent](eng
     }
   }
 
+  override def inputMany(data: Seq[String]): Unit = {
+    val aliases = params.indicators.flatMap { ip =>
+      ip.aliases.getOrElse(Seq(ip.name))
+    }
+    val (events, items) = data.view.foldLeft[(List[UREvent], List[UREvent])]((List.empty, List.empty)) { case (acc@(el, il), jsonEvent) =>
```
Comment (pferrel) [inline rest-server/engines/src/main/scala/com/actionml/engines/ur/URDataset.scala:163]: This is some pattern matching that I do not understand: `case (acc@(el, il)` what does it do? what is it called?

@andreyshikov

Comment (andreyshikov) [reply rest-server/engines/src/main/scala/com/actionml/engines/ur/URDataset.scala:163]: I'll answer in slack

File: rest-server/core/src/main/scala/com/actionml/core/engine/Engine.scala (hunk 2 above)
Comment (andreyshikov) [reply rest-server/core/src/main/scala/com/actionml/core/engine/Engine.scala:132]: It's not about call overhead, it's a sign that we can use a bulk insert that gives us a performance boost

**Comment by pferrel** [comment]
@andreyshikov these conflicts need your resolution. these are the same as the conflicts in #169

**Pull request merged**